src/main/java/bean/ui/Ui.java

```diff
@@ -0,0 +1,140 @@
+package bean.ui;
+
+import bean.task.Task;
+
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The Ui class handles user interactions, including displaying messages and reading user input.
+ */
+public class Ui {
+    private Scanner scanner;
+
+    /**
+     * Initializes a new Ui instance.
+     */
+    public Ui() {
+        this.scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Reads and returns user input as a trimmed string.
+     *
+     * @return The user's input.
+     */
+    public String getUserInput() {
+        return scanner.nextLine().trim();
+    }
+
+    /**
+     * Displays a greeting message to the user.
+     */
+    public void showGreeting() {
+        System.out.println("______________________________");
```

Comment:
I see that you use the divider quite often in your code. You might consider adding a method which prints the divider as it would save you time from copy and pasting the methods:

```
private void printDivider() {
        System.out.println("______________________________");
}
```